functional/stateful-types.md

```diff
@@ -0,0 +1,269 @@
+# Stateful Types: Generics as Type Classes
+
+## Description
+
+Rust's functional roots allow for it to be more expressive in the type system
```

Comment:
This seems weird.

> Rust's functional roots allows it to be ...

Will this be better? I still find that weird though.

Reply:
I have rewritten this sentence completely.